### DISCUSSION TITLE
What's the difference between Deploy Keys and SSH keys?

### DISCUSSION THREAD

**Discussion by lwcorp** [community/community#40788] [New to GitHub]
**Repositories** have settings for **Deploy Keys**, while **profiles** have settings for **SSH Keys**. Could you please explain the differences?

The specific goal is to connect a hosting account's cPanel to a GitHub private repository, but it's a general question.

From **Repositories' Settings**:
![image](https://user-images.githubusercontent.com/1773306/205519012-22a336ef-2987-44c5-94e8-3237b423099f.png)
From **Profiles' Settings**:
![image](https://user-images.githubusercontent.com/1773306/205519018-c4cebb29-49cc-4923-8c5b-c30747fc17ad.png)

**Comment by mohamedzhioua** [comment]
@lwcorp Your ssh account key does identify you, and allows you to interact with GitHub as you (meaning for instance; you can push to your own repo)

Your ssh deploy key, as mentioned in "[Read-only deploy keys](https://github.com/blog/2024-read-only-deploy-keys)" are keys with read-only access.
It does not authenticate you.

A deploy key is an SSH key that is stored on your server and grants access to a single GitHub repository. They are often used to clone repositories during deploys or continuous integration runs.
Deploys sometimes involve merging branches and pushing code, so deploy keys have always allowed both read and write access. Because write access is undesirable in many cases, you now have the ability to create deploy keys with read-only access.

You can create a read/write deploy key, but its ability to be read-only is in contrast with an account SSH key: the latter (SSH key) will always be for read/write access to your repositories.

You find the same purpose for [deploy keys at BitBucket](https://confluence.atlassian.com/bitbucket/use-deployment-keys-294486051.html):

A deployment key grants read-only access to a public or private repository. With a deployment key a user or a process can pull or clone a repository over SSH. Deployment keys have the following features and limitations:

Deployment keys do not apply to your plan limit.
You can add the same deployment key to multiple repositories.
The deployment key must be unique — it cannot also be associated with an account.

**Reply by Ahalll62199** [reply]
Vjj

**Reply by lwcorp** [reply]
Revoking answer as you didn't follow through.
Like you mentioned yourself, deploy key can be write and not just read.
So the way I see it , SSH keys give access to your entire repositories and use your own username for it, while Deploy key are for specific repositories and not through own username, am I right?

**Comment by lwcorp** [comment]
Thanks! So please help me out in #40790 - how do I connect my cPanel Git Version Control to a private GitHub repository with either of Deployed Keys or SSH Keys?

**Comment by mohamedzhioua** [comment]
@lwcorp  you can accept my answer so it count for me and i'll take a look in #40790 and try to help you
